src/basic/efivars.c

```diff
@@ -244,10 +244,16 @@ int efi_set_variable_string(sd_id128_t vendor, const char *name, const char *v)
 }
 
 bool is_efi_boot(void) {
-        if (detect_container() > 0)
-                return false;
+        static int cache = -1;
 
-        return access("/sys/firmware/efi/", F_OK) >= 0;
+        if (cache < 0) {
+                if (detect_container() > 0)
+                        cache = false;
+                else
+                        cache = access("/sys/firmware/efi/", F_OK) >= 0;
```

Comment:
is just accessing ths directory itself rate-limited too or only reading the actual variables? If the second than caching here seems redundant.

Reply:
well, this felt like something we might end up calling all the time, as it is the first thing we basically do in pretty much *any* EFI operation… this is hence not an attempt to optimize the kernel's ratelimit away, but just to avoid the syscall.

(it's much nicer to look at strace outputs btw, if this is cached, as the access() call only appears once that way)

Reply:
sounds good to me